src/utils.jl

```diff
@@ -21,25 +21,55 @@ equal probability is assigned to each entry in the support.
 
 # Examples
 ```julia
-using KernelFunctions
-# rows correspond to samples
-μ = discretemeasure(RowVecs(rand(7,3)), normalize!(rand(10),1))
 
-# columns correspond to samples, each with equal probability
-ν = discretemeasure(ColVecs(rand(3,12)))
+μ = discretemeasure(rand(10), normalize!(rand(10),1))
+
+# each entry has equal probability
+ν = discretemeasure(rand(3))
 ```
 
 !!! note
     If `support` is a 1D vector, the constructed measure will be sorted,
     e.g. for `mu = discretemeasure([3, 1, 2],[0.5, 0.2, 0.3])`, then
     `mu.support` will be `[1, 2, 3]` and `mu.p` will be `[0.2, 0.3, 0.5]`.
+    Also, avoid passing 1D distributions as something like `[[3],[1],[2]]`
+    since this will be dispatched to the multivariate case instead
+    of the univariate case for which the algorithm is more efficient.
 """
 function discretemeasure(
     support::AbstractVector{<:Real},
     probs::AbstractVector{<:Real}=fill(inv(length(support)), length(support)),
 )
     return DiscreteNonParametric(support, probs)
 end
+"""
+    discretemeasure(
+        support::AbstractVector,
+        probs::AbstractVector{<:Real}=fill(inv(length(support)), length(support)),
+    )
+
+Construct a finite discrete probability measure with `support` and corresponding
+`probabilities`. If the probability vector argument is not passed, then
+equal probability is assigned to each entry in the support.
+
+# Examples
+```julia
+using KernelFunctions
+# rows correspond to samples
+μ = discretemeasure(RowVecs(rand(7,3)), normalize!(rand(10),1))
+
+# columns correspond to samples, each with equal probability
+ν = discretemeasure(ColVecs(rand(3,12)))
+```
+
+!!! note
+    If `support` is a 1D vector, the constructed measure will be sorted,
+    e.g. for `mu = discretemeasure([3, 1, 2],[0.5, 0.2, 0.3])`, then
+    `mu.support` will be `[1, 2, 3]` and `mu.p` will be `[0.2, 0.3, 0.5]`.
+    Also, avoid passing 1D distributions as something like `[[3],[1],[2]]`
+    since this will be dispatched to the multivariate case instead
+    of the univariate case for which the algorithm is more efficient.
+"""
```

Comment:
I think it's a bit redundant with two docstrings since in the end both methods are doing the same thing - you give it a vector of the support and corresponding probabilities (which default to a vector of equal probabilities). If we want to document the multivariate case explicitly I think it would be better and more concise to integrate the example in the other docstring.

Reply:
Ok.